### PR TITLE
Do not leak etcd clients when generating new VNIs

### LINKED ISSUE #55: Leak of connection to ETCD
When run for a long time, the process starts to leak sockets, visible in 

```
ls /proc/$pid/fd
```

Details can be found with

```
sudo lsof -i -a -p $pid
```

After investigating one case, etcd connections are lost over time

### PR DESCRIPTION
**Pull request by Soulou**
Fixes #55

### CODE REVIEW

File: idmanager/idmanager.go [hunk 1]
```diff
@@ -3,51 +3,71 @@ package idmanager
...
+
 func New(c *config.Config, s store.Store, name string, field string, prefix string) Manager {
 	return &manager{config: c, store: s, field: field, name: name, prefix: prefix}
 }
 
-func (m *manager) Lock(ctx context.Context) error {
+func (m *manager) Lock(ctx context.Context) (Lock, error) {
+	var l Lock
+
 	client, err := etcd.NewClient()
```
Comment (Soulou) [inline idmanager/idmanager.go:44]: Here we were creating a new client, which was never Close(), so we were leaking connections to etcd each time a new ID was generated

Comment (Soulou) [reply idmanager/idmanager.go:44]: The goal of this PR is to close correctly the client after usage

Comment (johnsudaar) [reply idmanager/idmanager.go:44]: Thanks !

File: idmanager/idmanager.go [hunk 2]
```diff
@@ -3,51 +3,71 @@ package idmanager
...
 )
 
 type Manager interface {
-	Lock(context.Context) error
+	Lock(context.Context) (Lock, error)
 	Generate(context.Context) (int, error)
-	Unlock(context.Context) error
+}
+
+type Lock interface {
```
Comment (EtienneM) [inline idmanager/idmanager.go:20]: Name of the interface seems odd. I would call it `Unlocker`

File: idmanager/idmanager.go [hunk 3]
```diff
@@ -3,51 +3,71 @@ package idmanager
...
-func (m *manager) Unlock(ctx context.Context) error {
-	if m.lock == nil {
+func (l lock) Unlock(ctx context.Context) error {
+	if l.resourceLock == nil {
 		return errors.New("not locked")
 	}
-	return m.lock.Release()
+	err := l.resourceLock.Release()
+	if err != nil {
+		return errors.Wrapf(err, "fail to release etcd lock")
```
Comment (EtienneM) [inline idmanager/idmanager.go:64]: Shouldn't you just log the error but keep executing the function to close the `lockingBackend`, even though there has been an error when releasing the `resourceLock`?

Comment (Soulou) [reply idmanager/idmanager.go]: Fixed

File: idmanager/idmanager.go (hunk 2 above)
Comment (Soulou) [reply idmanager/idmanager.go]: Changed

**Pull request merged**